tensorboard/webapp/widgets/data_table/header_cell_component_test.ts

```diff
@@ -96,4 +104,42 @@ describe('header cell', () => {
     fixture.debugElement.query(By.css('.cell')).nativeElement.click();
     expect(headerClickedSpy).toHaveBeenCalledOnceWith('run');
   });
+
+  describe('delete column button', () => {
+    it('emits removeColumn event when delete button clicked', () => {
+      const fixture = createComponent({});
+      fixture.componentInstance.hparamsEnabled = true;
+      fixture.detectChanges();
+
+      fixture.debugElement
+        .query(By.directive(HeaderCellComponent))
+        .componentInstance.deleteButtonClicked.subscribe();
+      fixture.debugElement
+        .query(By.css('.delete-icon'))
+        .triggerEventHandler('click', {});
+
+      expect(deleteButtonClickedSpy).toHaveBeenCalledOnceWith({
+        name: 'run',
+        displayName: 'Run',
+        type: ColumnHeaderType.RUN,
+        enabled: true,
+      });
+    });
+
+    it('renders delete button when hparam flag is on', () => {
+      const fixture = createComponent({});
+      fixture.componentInstance.hparamsEnabled = true;
+      fixture.detectChanges();
+
+      expect(fixture.debugElement.query(By.css('.delete-icon'))).toBeTruthy();
+    });
+
+    it('does not render delete button when hparam flag is off', () => {
+      const fixture = createComponent({});
+      fixture.componentInstance.hparamsEnabled = true;
+      fixture.detectChanges();
```

Comment:
You can consider putting these three lines into beforeEach and additional settings in the third test. Eventually this test will be removed when we turn on the feature flag.

some pseudo code like
beforeEach {
      const fixture = createComponent({});
      fixture.componentInstance.hparamsEnabled = true;
      fixture.detectChanges();
}

it () {
      fixture.componentInstance.hparamsEnabled = false;
      fixture.detectChanges();
}